library/CodeMonkeysRu/GCM/Response.php

```diff
@@ -63,26 +108,72 @@ public function __construct(Message $message, $responseBody)
         $this->failure = $data['failure'];
         $this->success = $data['success'];
         $this->canonicalIds = $data['canonical_ids'];
+        $this->existsInvalidDataKey = false;
+        $this->existsMismatchSenderId = false;
         $this->results = array();
+        
         foreach ($message->getRegistrationIds() as $key => $registrationId) {
-            $this->results[$registrationId] = $data['results'][$key];
+            $result = $data['results'][$key];
+            if (isset($result['error'])) {
+	        switch ($result['error']) {
+		    case "InvalidDataKey":
+		        $this->existsInvalidDataKey = true;
+			break;
+		    case "MismatchSenderId":
+		        $this->existsMismatchSenderId = true;
+			break;
+		    default:
+		        break;
+	        }
+	    }
+	    $this->results[$registrationId] = $result; 
         }
+        $result = null;
```

Comment:
We use that variable in the ```foreach```, and if a developer wants to add to the constructor and use the variable ```$result```, it should be empty.